### PR TITLE
Add achievements to THoT

### PR DESCRIPTION
**Pull request by Gothyoba**
The following achievements have been added:

- Gryphon Hunter: Kill the Gryphon leader in High Pass

- Conqueror of the Forest: Defeat an enemy leader in Forbidden Forest

- New Thursagan: Complete The Underlevels on challenging difficulty

### CODE REVIEW

File: data/campaigns/The_Hammer_of_Thursagan/achievements.cfg [hunk 1]
```diff
@@ -1 +1,27 @@
 #textdomain wesnoth-thot
+
+[achievement_group]
+    display_name=_"The Hammer of Thursagan"
+    content_for=the_hammer_of_thursagan
+    [achievement]
+        id="gryphon_hunter"
+        name=_"Gryphon Hunter"
+        description=_"Kill the Gryphon leader in High Pass"
```
Comment (knyghtmare) [inline data/campaigns/The_Hammer_of_Thursagan/achievements.cfg:9]: This is kind of spoilery

File: data/campaigns/The_Hammer_of_Thursagan/scenarios/09_The_Underlevels.cfg [hunk 2]
```diff
@@ -1294,6 +1294,13 @@
             result=victory
             carryover_report=no
         [/endlevel]
+
+        #ifdef HARD
```
Comment (Wedge009) [inline data/campaigns/The_Hammer_of_Thursagan/scenarios/09_The_Underlevels.cfg:1298]: I think the checks are complaining about the indentation here - suggest no indent for the `#if` statements and align the WML according to how it should be if there were no `#if` statements.

**Comment by CelticMinstrel** [comment]
The two added `[message]` blocks seem a bit… unnecessary?

File: data/campaigns/The_Hammer_of_Thursagan/achievements.cfg (hunk 1 above)
Comment (Gothyoba) [reply data/campaigns/The_Hammer_of_Thursagan/achievements.cfg:9]: This is now a hidden achievement.

File: data/campaigns/The_Hammer_of_Thursagan/scenarios/09_The_Underlevels.cfg (hunk 2 above)
Comment (Gothyoba) [reply data/campaigns/The_Hammer_of_Thursagan/scenarios/09_The_Underlevels.cfg:1298]: Still not working. I'm somewhat confused here. Is there something about WML indentation I'm missing?

**Comment by Gothyoba** [comment]
> The two added `[message]` blocks seem a bit… unnecessary?

I removed the gryphon one, I agree it's unnecessary. Personally I like the one in Forbidden Forest.

File: data/campaigns/The_Hammer_of_Thursagan/scenarios/09_The_Underlevels.cfg (hunk 2 above)
Comment (Wedge009) [reply data/campaigns/The_Hammer_of_Thursagan/scenarios/09_The_Underlevels.cfg]: This particular instance is resolved.

**Comment by Wedge009** [comment]
I can't see it in the differences, but the CI check seems to think there are additional spaces on a blank line in the change to `04_High_Pass.cfg`

**Comment by Gothyoba** [comment]
> I can't see it in the differences, but the CI check seems to think there are additional spaces on a blank line in the change to `04_High_Pass.cfg`

Thanks! Looks like it's finally fixed!

**Comment by Wedge009** [comment]
Great. Looks like it's a fault (or feature) of the GitHub difference viewer. When I look at your original commit, the white-space doesn't show in the difference. But when I go to the actual file, the white-space shows. Something to keep in mind for the future, then.

**Comment by Gothyoba** [comment]
Should this get a changelog entry? Something like Added Achievements to THoT? I'm not sure what the requirements are for that. A lot of stuff never gets added to the changelog.

**Comment by Pentarctagon** [comment]
It'd be good to add. A lot of stuff doesn't get added not because it shouldn't get added but because the developer forgot to add it.

**Comment by Wedge009** [comment]
As something visible to the player, I'd certainly advise adding a change log entry.

**Comment by Gothyoba** [comment]
I added a changelog entry!

**Comment by Gothyoba** [comment]
Does this need any further review?

**Comment by Gothyoba** [comment]
Does this need any further review?

**Comment by Pentarctagon** [comment]
Sorry, it needs Dalas's approval - I've added him as a reviewer.

**Review by Dalas121 — APPROVED** [review]
Assuming these have been tested and work correctly, nothing stands out to me as problematic. Glad to see achievements!

**Review by knyghtmare — APPROVED** [review]

**Pull request merged**